Tests/MongoSwiftSyncTests/MongoCollectionTests.swift

```diff
@@ -589,6 +589,105 @@ final class MongoCollectionTests: MongoSwiftTestCase {
             .to(throwError(errorType: MongoError.InvalidArgumentError.self))
     }
 
+    func testHintInPreviousVersion() throws {
+        try withTestNamespace { client, _, collection in
+            let requirements = TestRequirement(
+                maxServerVersion: ServerVersion(major: 4, minor: 2)
+            )
+            let unmetRequirement = try client.getUnmetRequirement(requirements)
+            guard unmetRequirement == nil else {
+                printSkipMessage(testName: self.name, unmetRequirement: unmetRequirement!)
+                return
+            }
+
+            // Drivers must raise an error for servers < 4.2.
+            let invalidOpts1 = FindOneAndReplaceOptions(hint: ["nonexistant_index": 1])
```

Comment:
Because this test would also currently generate errors even on 4.2+ due  to the index not existing, it makes it slightly unclear exactly what is being verified on < 4.2. If we instead modify the case such that it'd cause errors _only_ on 4.2, then it becomes more clear what is being evaluated. To that end, I think it would be marginally better to run this test with a valid hint specified (i.e. an index that does exist) so that we can ensure the driver is failing for the right reason.

Reply:
done